### PR TITLE
feat: (issue 2429) read gRPC hiero-correlation-id header and prefix publisher handler logs

### PR DESCRIPTION
**Pull request by a-saksena**
## Related Issue(s)
 Use keywords `Fix, Fixes, Fixed, Close, Closes, Closed, Resolve, Resolves, Resolved`
to connect an issue to be closed by this pull request.

### CODE REVIEW

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 1]
```diff
@@ -207,9 +227,9 @@ public void sendAcknowledgement(final long newLastAcknowledgedBlockNumber) {
...
             metrics.blockAcknowledgementsSent.increment(); // @todo(1415) add label
 
-            final String ackMessage = "Sent acknowledgement for block {0,number,#} from handler {1}";
-            final String traceMessage =
-                    "metric-end-to-end-latency-by-block-end block={0,number,#} nsTimestamp={1,number,#} handlerId={2}";
+            final String ackMessage = ctx() + "Sent acknowledgement for block {0,number,#} from handler {1}";
+            final String traceMessage = ctx()
+                    + "metric-end-to-end-latency-by-block-end block={0,number,#} nsTimestamp={1,number,#} handlerId={2}";
             LOGGER.log(TRACE, traceMessage, newLastAcknowledgedBlockNumber, System.nanoTime(), handlerId);
             LOGGER.log(TRACE, ackMessage, newLastAcknowledgedBlockNumber, handlerId);
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Another example; don't add the ID to the string, let the logger add it _when needed_.
Also, the way you added it to the trace message breaks log metrics, you have to add at the end with a key=value pattern.

```suggestion
            final String ackMessage = "{2} Sent acknowledgement for block {0,number,#} from handler {1}";
            final String traceMessage = "metric-end-to-end-latency-by-block-end block={0,number,#} nsTimestamp={1,number,#} handlerId={2} correlationId={3}";
LOGGER.log(TRACE, traceMessage, newLastAcknowledgedBlockNumber, System.nanoTime(), handlerId, correlationIdPrefix);
LOGGER.log(TRACE, ackMessage, newLastAcknowledgedBlockNumber, handlerId, correlationIdPrefix);
```

**Review by jsync-swirlds — CHANGES_REQUESTED** [review]
The way this was added isn't good and breaks some important metrics.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java [hunk 2]
```diff
@@ -125,11 +125,129 @@ void testNullMetricsHolder() {
...
+
+        /**
+         * This test aims to assert that the constructor of {@link PublisherHandler} throws a
+         * {@link NullPointerException} when the correlation ID is null.
+         */
+        @Test
+        @DisplayName("Test constructor with null correlation ID")
+        void testNullCorrelationId() {
+            assertThatNullPointerException().isThrownBy(() -> {
+                new PublisherHandler(validNextId, validReplyPipeline, validMetricsHodler, validPublisherManager, null);
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java]: We generally want the _opposite_ of this.
Exceptions are a lot harder to handle than just not including a trace value, so it's better to just silently leave the ID value empty in the handler rather than throw a NPE for data that a publisher might not send.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java [hunk 3]
```diff
@@ -177,10 +192,26 @@ public List<Class<? extends Record>> configDataTypes() {
...
+    ///
+    /// @param correlationId the raw value from the gRPC header
+    /// @return the value unchanged if within the limit, or truncated to [#MAX_CORRELATION_ID_LENGTH]
+    @NonNull
+    static String truncateCorrelationId(@NonNull final String correlationId) {
+        if (correlationId.length() > MAX_CORRELATION_ID_LENGTH) {
+            return correlationId.substring(0, MAX_CORRELATION_ID_LENGTH);
+        }
+        return correlationId;
     }
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java]: Rather than making this "NonNull", we should probably just allow and handle null correlation ID values.

```suggestion
    /// Truncates the given correlation ID to [#MAX_CORRELATION_ID_LENGTH] characters.
    ///
    /// Protects against oversized header values that could inflate every log line.
    /// If the value exceeds the limit, it is truncated and a warning is logged.
    ///
    /// @param correlationId the raw value from the gRPC header
    /// @return the value unchanged if within the limit, or truncated to [#MAX_CORRELATION_ID_LENGTH]
    static String truncateCorrelationId(final String correlationId) {
        if (correlationId != null && correlationId.length() > MAX_CORRELATION_ID_LENGTH) {
            return correlationId.substring(0, MAX_CORRELATION_ID_LENGTH);
        }
        return correlationId;
}
```

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 4]
```diff
@@ -156,13 +173,13 @@ public void onSubscribe(@NonNull final Subscription subscription) {
     @Override
     public void onNext(@NonNull final PublishStreamRequestUnparsed request) {
         try {
-            LOGGER.log(TRACE, "Handler {0} received request", handlerId);
+            LOGGER.log(TRACE, ctx() + "Handler {0} received request", handlerId);
             processNextRequestUnparsed(request);
-            LOGGER.log(TRACE, "Handler {0} finished processing request", handlerId);
+            LOGGER.log(TRACE, ctx() + "Handler {0} finished processing request", handlerId);
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Agree to use string interpolation instead of concatenation.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 5]
```diff
@@ -103,21 +107,34 @@ public final class PublisherHandler implements Pipeline<PublishStreamRequestUnpa
...
             @NonNull final MetricsHolder handlerMetrics,
-            @NonNull final StreamPublisherManager manager) {
+            @NonNull final StreamPublisherManager manager,
+            @NonNull final String correlationId) {
         handlerId = nextId;
         replies = Objects.requireNonNull(replyPipeline);
         metrics = Objects.requireNonNull(handlerMetrics);
         publisherManager = Objects.requireNonNull(manager);
+        final String validatedCorrelationId = Objects.requireNonNull(correlationId);
+        this.correlationIdPrefix = validatedCorrelationId.isEmpty() ? "" : "[" + validatedCorrelationId + "] ";
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Agree that the id can come in as Null.
However, it makes the log much clearer with the [ ].

[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: There are some forms that cannot include the `[]`, so I've been adding that _when needed_ in the log message in these examples.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java (hunk 2 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java:141]: I will write another explicit test to test that Null is a valid value and NPE is not thrown because of it being null.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java [hunk 6]
```diff
@@ -183,12 +183,12 @@ void setup() {
             // Create a response pipeline to handle the responses from the first publisher handler.
             responsePipeline = new TestResponsePipeline();
             // Create the first publisher handler and add it to the manager.
-            publisherHandler = toTest.addHandler(responsePipeline, sharedHandlerMetrics);
+            publisherHandler = toTest.addHandler(responsePipeline, sharedHandlerMetrics, "");
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java]: One approach is to use a `@ParameterizedTest` to cover `null`, `""` and non-empty for the same test

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Good start including passing through call chain and pre-computing correlationIdPrefix, character truncations and test additions.

@jsync-swirlds covered most of the cases I'm concerned about with respect to perf and breaking log layouts for other telemetry.
Also kindly add a PR description that makes it easier for reviewers to understand what they are reviewing and the impact

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 7]
```diff
@@ -103,21 +107,34 @@ public final class PublisherHandler implements Pipeline<PublishStreamRequestUnpa
...
     }
 
+    /// Returns the pre-computed log prefix for this handler.
+    ///
+    /// Returns `"[correlationId] "` if a correlation ID was provided at connection time,
+    /// or `""` if none was present. Pre-computed in the constructor to avoid allocations on hot paths.
+    private String ctx() {
+        return correlationIdPrefix;
+    }
+
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java (hunk 1 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 8]
```diff
@@ -330,8 +350,8 @@ private void handleBlockItemsRequest(
             currentStreamingBlockNumber.set(blockNumber);
             // this means that we are starting a new block, so we can
             // update the current streaming block number
-            final String traceMessage =
-                    "metric-end-to-end-latency-by-block-start block={0,number,#} nsTimestamp={1,number,#} handlerId={2}";
+            final String traceMessage = ctx()
+                    + "metric-end-to-end-latency-by-block-start block={0,number,#} nsTimestamp={1,number,#} handlerId={2}";
             currentStreamingBlockHeaderReceivedTime = System.nanoTime();
             LOGGER.log(TRACE, traceMessage, blockNumber, currentStreamingBlockHeaderReceivedTime, handlerId);
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 9]
```diff
@@ -394,8 +415,9 @@ private void handleEndStreamRequest(final EndStream endStream) {
         } else {
             LOGGER.log(
                     INFO,
-                    "Handler %d received an invalid EndStream request with code %s. %s"
-                            .formatted(handlerId, code, earliestAndLatestBlockNumbers));
+                    ctx()
+                            + "Handler %d received an invalid EndStream request with code %s. %s"
+                                    .formatted(handlerId, code, earliestAndLatestBlockNumbers));
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 10]
```diff
@@ -544,16 +567,17 @@ private boolean sendResponse(final PublishStreamResponse response) {
             // at debug rather than emitting noise in the logs.
             // Also, this confuses everyone, they all see this debug log and
             // assume the node crashed, so we must not print a stack trace.
-            final String messageFormat = "Publisher closed the connection unexpectedly for client %d: %s";
+            final String messageFormat = ctx() + "Publisher closed the connection unexpectedly for client %d: %s";
             final String exceptionMessage = e.getCause() != null ? e.getCause().getMessage() : e.getMessage();
             final String message = messageFormat.formatted(handlerId, exceptionMessage);
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 11]
```diff
@@ -753,20 +784,21 @@ private void shutdown() {
...
+            LOGGER.log(TRACE, ctx() + "Handler {0} issued onComplete/closeConnection", handlerId);
         }
     }
 
     /// Check if we have a block in progress. The parameter should be used to supply the
     /// [#currentStreamingBlockNumber] value.
     private boolean isCurrentlyMidBlock(final long blockInProgress) {
         if (blockInProgress != currentStreamingBlockNumber.get()) {
-            LOGGER.log(DEBUG, "Ending mid block, but block number does not match.");
+            LOGGER.log(DEBUG, ctx() + "Ending mid block, but block number does not match.");
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/module-info.java [hunk 12]
```diff
@@ -19,6 +19,7 @@
     requires org.hiero.block.node.app.config;
     requires org.hiero.block.node.base;
     requires com.github.spotbugs.annotations;
+    requires java.logging;
```
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/module-info.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (a-saksena) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java (hunk 2 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java:141]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java (hunk 3 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java:216]: Review comment incorporated in the latest code.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 13]
```diff
@@ -207,11 +218,19 @@ public void sendAcknowledgement(final long newLastAcknowledgedBlockNumber) {
...
             metrics.blockAcknowledgementsSent.increment(); // @todo(1415) add label
 
-            final String ackMessage = "Sent acknowledgement for block {0,number,#} from handler {1}";
-            final String traceMessage =
-                    "metric-end-to-end-latency-by-block-end block={0,number,#} nsTimestamp={1,number,#} handlerId={2}";
-            LOGGER.log(TRACE, traceMessage, newLastAcknowledgedBlockNumber, System.nanoTime(), handlerId);
-            LOGGER.log(TRACE, ackMessage, newLastAcknowledgedBlockNumber, handlerId);
+            LOGGER.log(
+                    TRACE,
+                    "metric-end-to-end-latency-by-block-end block={0,number,#} nsTimestamp={1,number,#} handlerId={2} correlationId=[{3}]",
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Here we must not include the `[]` characters; it breaks parsing in Loki.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java [hunk 14]
```diff
@@ -330,18 +358,26 @@ private void handleBlockItemsRequest(
             currentStreamingBlockNumber.set(blockNumber);
             // this means that we are starting a new block, so we can
             // update the current streaming block number
-            final String traceMessage =
-                    "metric-end-to-end-latency-by-block-start block={0,number,#} nsTimestamp={1,number,#} handlerId={2}";
             currentStreamingBlockHeaderReceivedTime = System.nanoTime();
-            LOGGER.log(TRACE, traceMessage, blockNumber, currentStreamingBlockHeaderReceivedTime, handlerId);
+            LOGGER.log(
+                    TRACE,
+                    "metric-end-to-end-latency-by-block-start block={0,number,#} nsTimestamp={1,number,#} handlerId={2} correlationId=[{3}]",
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Same here, this is a metric value, so we must not add extra characters that cause parsing issues.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java [hunk 15]
```diff
@@ -1392,6 +1392,60 @@ void testHandlePersistedNotificationFailedPersistence() {
                 assertThat(responsePipeline.getClientEndStreamCalls().get()).isEqualTo(0);
             }
 
+            /// Verifies that a failed [PersistedNotification] (succeeded = false)
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java:1395]: These coverage tests will probably conflict with @ata-nas latest changes; so we should be careful about what order we merge the two PRs.

File: tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh [hunk 16]
```diff
@@ -8,7 +8,7 @@ set -euo pipefail
 
 SDK_DIR=""
 TCK_DIR=""
-TEST_FILE="src/tests/crypto-service/test-transfer-hbar-transaction.ts src/tests/crypto-service/test-account-create-transaction.ts src/tests/crypto-service/test-account-update-transaction.ts src/tests/token-service/test-token-create-transaction.ts src/tests/token-service/test-token-update-transaction.ts src/tests/topic-service/test-topic-create-transaction.ts src/tests/topic-service/test-topic-update-transaction.ts src/tests/contract-service/test-contract-delete-transaction.ts"
```
Comment (jsync-swirlds) [inline tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh:11]: Why did we remove these?

File: tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh [hunk 17]
```diff
@@ -90,17 +90,12 @@ fi
...
 cp .env.custom_node .env
-if [[ "${TEST_FILE}" == "all" ]]; then
-  echo "Running all TCK tests (test:serial)"
-  npm run test:serial
-else
-  echo "Running TCK tests: $TEST_FILE"
-  # shellcheck disable=SC2086
-  # Word splitting is intentional — TEST_FILE may contain multiple space-separated file paths
-  npm run test:file $TEST_FILE
-fi
```
Comment (jsync-swirlds) [inline tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh:103]: Similar concern, this seems to remove the option to run "all" tests.

cc: @Nana-EC

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java [hunk 18]
```diff
@@ -1946,6 +2067,111 @@ void testOnNextEndStreamTooFarBehindNoNotification() {
                         manager.getBlockMessagingFacility().getSentNewestBlockKnownToNetworkNotifications();
                 assertThat(sentNotifications).hasSize(0);
             }
+
```
Comment (jsync-swirlds) [inline block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java:2070]: These should also be examined by @ata-nas to ensure they aren't conflicting with his recent changes.

**Review by jsync-swirlds — DISMISSED** [review]
A couple changes needed for the metric logs.
These are log statements we use Loki to parse into metrics because they're cross-plugin telemetry data we cannot easily turn into useable metrics otherwise.

File: tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh (hunk 17 above)
Comment (Nana-EC) [reply tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh:103]: Yeah, please revert all changes in this file for this PR.
If we want to streamline anything here let's do it separately so this PR doesn't get held up

File: tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh (hunk 16 above)
Comment (Nana-EC) [reply tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh:11]: Please revert

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java (hunk 14 above)
Comment (Nana-EC) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: It's a gap in our documentation but basically anywhere you see the format <key>=<value> is an instance of logs that Loki can convert into telemetry and we can use to create adhoc telemetry.
As such we don't want to break those but you can add key-value pairs to them

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java [hunk 19]
```diff
@@ -122,6 +127,16 @@ public Pipeline<? super Bytes> open(
         final int maxMessageSize =
                 context.configuration().getConfigData(ServerConfig.class).maxMessageSizeBytes() - 16384;
 
+        final String rawCorrelationId = options.metadata().getOrDefault("hiero-correlation-id", "");
+        if (rawCorrelationId.length() > MAX_CORRELATION_ID_LENGTH) {
```
Comment (Nana-EC) [inline block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java]: I think this logic is repeated in truncateCorrelationId so you can get rid of it and just call the method

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java [hunk 20]
```diff
@@ -122,6 +127,16 @@ public Pipeline<? super Bytes> open(
...
+            LOGGER.log(
+                    System.Logger.Level.WARNING,
+                    "Received hiero-correlation-id header exceeds {0} characters and will be truncated: {1}",
```
Comment (Nana-EC) [inline block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java]: May also be worth logging the size that came in so it's clear how much over the limit we're observing

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Some mote items.
We should scope this to the necessary changes needed but also watch out for the value pairs in logs, those are logs that Loki can turn into telemetry that we specifcally formatted for

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java (hunk 13 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Incorporated. This should be fixed now.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java (hunk 14 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java]: Incorporated. This should be fixed now.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java (hunk 15 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/LiveStreamPublisherManagerTest.java:1395]: Will coordinate with Atanas.

File: tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh (hunk 17 above)
Comment (a-saksena) [reply tools-and-tests/scripts/solo-e2e-test/scripts/solo-tck-sdk-run.sh:103]: solo-tck-sdk-run.sh file changes got committed by mistake last time. Reverting.

File: block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java (hunk 18 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/test/java/org/hiero/block/node/stream/publisher/PublisherHandlerTest.java:2070]: Will coordinate with Atanas.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java (hunk 19 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java]: Incorporated. This should be fixed now.

File: block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java (hunk 20 above)
Comment (a-saksena) [reply block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/StreamPublisherPlugin.java]: Incorporated. This should be fixed now.

**Review by jsync-swirlds — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
